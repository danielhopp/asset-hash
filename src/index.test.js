import hashFile from "./index"

test("Encode text", async () => {
  const hash = await hashFile("./src/fixtures/text.md")
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF", async () => {
  const hash = await hashFile("./src/fixtures/font.woff")
  expect(hash).toMatchSnapshot()
})

test("Encode PNG", async () => {
  const hash = await hashFile("./src/fixtures/image.png")
  expect(hash).toMatchSnapshot()
})


test("Encode text - SHA256", async () => {
  const hash = await hashFile("./src/fixtures/text.md", "sha256")
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF - SHA256", async () => {
  const hash = await hashFile("./src/fixtures/font.woff", "sha256")
  expect(hash).toMatchSnapshot()
})

test("Encode PNG - SHA256", async () => {
  const hash = await hashFile("./src/fixtures/image.png", "sha256")
  expect(hash).toMatchSnapshot()
})
