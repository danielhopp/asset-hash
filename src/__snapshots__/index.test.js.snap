// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class: Defaults - Base64 1`] = `"BtQzcA=="`;

exports[`Class: Defaults - Hex 1`] = `"06d43370"`;

exports[`Class: Defaults 1`] = `"eXxUvI"`;

exports[`Class: SHA256 - Base64 1`] = `"A9yvJrOL"`;

exports[`Class: SHA256 - Hex 1`] = `"03dcaf26"`;

exports[`Class: SHA256 1`] = `"eMjdHMlj"`;

exports[`Encode PNG - Base26 1`] = `"benehfw"`;

exports[`Encode PNG - SHA256 1`] = `"mDoTeFvG"`;

exports[`Encode PNG 1`] = `"XDOPW"`;

exports[`Encode WOFF - Base26 1`] = `"hirafra"`;

exports[`Encode WOFF - SHA256 1`] = `"IqPrPXfe"`;

exports[`Encode WOFF 1`] = `"fXQovA"`;

exports[`Encode text - Base26 1`] = `"hgehezu"`;

exports[`Encode text - SHA256 1`] = `"BpdHekEB"`;

exports[`Encode text - with invalid encoding crashes 1`] = `"Unknown base encoding base51!"`;

exports[`Encode text - with invalid hash crashes 1`] = `"Digest method not supported"`;

exports[`Encode text 1`] = `"fTNUMU"`;

exports[`FileName PNG 1`] = `"XDOPW.png"`;

exports[`FileName WOFF 1`] = `"fXQovA.woff"`;

exports[`FileName text 1`] = `"fTNUMU.md"`;
